backend/src/schema/resolvers/groups.ts

```diff
@@ -368,6 +368,64 @@ export default {
         session.close()
       }
     },
+    muteGroup: async (_parent, params, context, _resolveInfo) => {
+      const { id: groupId } = params
+      const userId = context.user.id
+      const session = context.driver.session()
+      const writeTxResultPromise = session.writeTransaction(async (transaction) => {
+        const transactionResponse = await transaction.run(
+          `
+          MATCH (group:Group { id: $groupId })
+          MATCH (user:User { id: $userId })
+          MERGE (user)-[m:MUTED]->(group)
+          SET m.createdAt = toString(datetime())
+          RETURN group { .* }
```

Comment:
User must be in group, you can't mute groups your not part of?!

Reply:
```
      muteGroup: and(isAuthenticated, isMemberOfGroup),
      unmuteGroup: and(isAuthenticated, isMemberOfGroup),
```

Reply:
I hate middlewares